File: Laba2-Python/Task6.py
import extensions
import numpy as np

def run():
    print("Завдання 6. У двовимірному масиві зберігається дохід 10 магазинів за кожен місяць року (кожному магазину відповідає один рядок). " +
        "\nНаписати програму, що визначає дохід обраного користувачем магазину за обраний квартал року.")

    matrix = np.random.randint(0, 1000, (10, 12))

    print("\nМатриця:\n")
    print_profit_matrix(matrix)

    ask_client_for_profit(matrix)

def print_profit_matrix(matrix):
    months = ["Січень", "Лютий", "Березень", "Квітень", "Травень", "Червень",
              "Липень", "Серпень", "Вересень", "Жовтень", "Листопад", "Грудень"]

    print(f"\n{'МІСЯЦІ': <13}", end="")
    for month in months:
        print(f"{month: >10}", end="")
    print()

    for i in range(matrix.shape[0]):
        print(f"Магазин #{i+1}: ", end="")
        for j in range(matrix.shape[1]):
            print(f"{matrix[i, j]: >10}", end="")
        print()

def ask_client_for_profit(matrix):
    print()

    is_asking = True

    while is_asking:
        shop_index = int(input("Введіть номер магазину: "))
        month_index = int(input("Введіть номер місяця: "))

        print(f"Дохід магазину #{shop_index} за місяць #{month_index}: {matrix[shop_index - 1, month_index - 1]}")

        answer = input("Продовжити? (y/n)")
        is_asking = answer.lower() == 'y'

File: Laba2/Task6.cs
using Extensions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PZPK_Labas.Laba2
{
    public static class Task6
    {
        public static void Run()
        {
            Console.WriteLine("Завдання 6. У двовимірному масиві зберігається дохід 10 магазинів за кожен місяць року (кожному магазину відповідає один рядок). " +
                "\nНаписати програму, що визначає дохід обраного користувачем магазину за обраний квартал року.");

            //int[,] matrix = ArrayHelper.InputMatrix(10, 12);
            int[,] matrix = ArrayHelper.GenerateMatrix(10, 12, 0, 1000);

            Console.WriteLine("\nМатриця:\n");
            PrintProfitMatrix(matrix);

            AskClientForProfit(matrix);
        }

        private static void PrintProfitMatrix(int[,] matrix)
        {
            // array of months            // array of months
            string[] months = new string[] { "Січень", "Лютий", "Березень", "Квітень",
                "Травень", "Червень", "Липень", "Серпень", "Вересень",
                "Жовтень", "Листопад", "Грудень"};
            // print months
            Console.Write("{0, -13}", "МІСЯЦІ");
            for (int i = 0; i < months.Length; i++)
                Console.Write("{0, 10}", months[i]);
            Console.WriteLine();
            // print profit

            for(int i = 0; i < matrix.GetLength(0); i++)
            {
                Console.Write("{0, -13}",$"Магазин #{i+1}: ");
                for(int j = 0; j < matrix.GetLength(1); j++)
                {
                    Console.Write("{0, 10}", matrix[i, j]);
                }
                Console.WriteLine();
            }
        }

        private static void AskClientForProfit(int[,] matrix)
        {
            Console.WriteLine();

            bool isAsking = true;

            while (isAsking)
            {
                int shopIndex, monthIndex;
                ConsoleInput.SafeIntInput("Введіть номер магазину", out shopIndex);
                ConsoleInput.SafeIntInput("Введіть номер місяця", out monthIndex);

                Console.WriteLine($"Дохід магазину #{shopIndex} за місяць #{monthIndex}: {matrix[shopIndex - 1, monthIndex - 1]}");

                Console.Write("Продовжити? (y/n)");
                isAsking = Console.ReadLine().Contains('y');
            }
        }
    }
}
